packages/ember-metal/lib/array.js

```diff
@@ -101,7 +101,7 @@ var filter = defineNativeShim(ArrayPrototype.filter, function(fn, context) {
   var length = this.length;
 
   for (i = 0; i < length; i++) {
-    if (this.hasOwnProperty(i)) {
+    if (typeof this.hasOwnProperty !== 'function' || this.hasOwnProperty(i)) {
```

Comment:
@stefanpenner  I replaced that code. But then Travis CI test failed for Safari. Not able to find the reason so i reverted back to same code which passed before. Now Travis CI test fails for IE9. Issue with travis?